docs/adr/adr_improved_frontend_dx.md

```diff
@@ -0,0 +1,55 @@
+# ADR: Improved Frontend Developer Experience
+Decision Deadline: As soon as possible
+
+Author: [Felipe](https://github.com/fhenrich33)
+
+## Status
+
+Proposed, Implementing
+
+## Context
+Frontend development experience in its current form is quite lacking. Imposes a lot of artificial difficulties, tooling doesn't work as expected, and adds friction to our FE contributors.
+
+We want to make FE development better.
+
+## Decision drivers
+
+1. We must make FE onboarding and development seamless.
+2. Implementation must play well with the current FE setup (React/Vite).
+3. Implementation must work well with our CI/CD pipeline.
+4. We must NOT interfere with BE tooling and development.
+
+## Considered options
+
+The only difference between proposed options is how to manage the monorepo, otherwise, there's a lot of overlap on proposed changes:
+- **[pnpm](https://pnpm.io/):** The "new" unwritten standard for package management--faster than both `npm` and `yarn`, and more forgiving on how it manages and caches `node_modules`. Will replace `yarn`.
+- **[husky](https://typicode.github.io/husky/) and [lint-staged](https://github.com/lint-staged/lint-staged):** Improved git hook configuration, also an unwritten standard in the FE ecosystem. Will replace our manual git hook setup.
+
+Current tooling like `Vite`, `TypeScript`, `ESLint`, and `Prettier` remains the same, and are expected to work flawlessly with these.
+
+Monorepo management:
+
+1. **[Turborepo](https://turbo.build/repo/docs):** Does a lot of lifting for us automatically, is content aware so it only works on relevant files, caches results of deterministic tasks locally ([and remotely](https://turbo.build/repo/docs/core-concepts/remote-caching), if we have a need for it in the future), and works seamlessly with current and suggested tooling.
```

Comment:
Lifting like what?

Reply:
Should have been more clear here, but everything: type checking, formatting, testing, building, etc.